### PR TITLE
Initial Code Release

### PR DESCRIPTION
**Pull request by LucZot**
This PR marks the initial release of the BUCTD codebase, providing code and instructions to tackle crowded pose estimation in humans and animals!

Overview of what this codebase includes:
- Environment installation instructions
- Instructions and code for training and testing BUCTD models
- Pre-trained checkpoints for human pose estimation benchmarks

Looking ahead, we plan to integrate our models into [DeepLabCut](https://github.com/DeepLabCut/DeepLabCut)! 

Please review the code and share your feedback. Thank you for your contributions!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,55 +1,140 @@
...
-
-- Stand-alone code will be made available by ICCV (Oct 2023)!
+<img src="media/BUCTD_fig1.png" width="600">
 
 - This code will also be integrated in [DeepLabCut](https://github.com/DeepLabCut/DeepLabCut)!
 
-### Code Acknowledgements
+### Installation 
+
+We developed and tested our models under ```python=3.8.10, pytorch=1.8.0, cuda=11.1```. Other versions might be working as fine.
```
Comment (MMathisLab) [inline README.md]: grammar

File: README.md [hunk 2]
```diff
@@ -1,55 +1,140 @@
...
+
+<details>
+  <summary>Instructions</summary>
+   
+   1. Clone this repo, and we'll call the directory that you cloned as ${BUCTD_ROOT}.
```
Comment (MMathisLab) [inline README.md]: grammar

File: README.md [hunk 3]
```diff
@@ -1,55 +1,140 @@
...
+
+   2. Install Pytorch and torchvision
+
+   Follow the instruction on https://pytorch.org/get-started/locally/.
+   ```sh
+   # an example:
+   conda install pytorch==1.8.0 torchvision==0.9.0 cudatoolkit=11.1 -c pytorch -c conda-forge
+   ```
+
+   3. Install other needed packages
```
Comment (MMathisLab) [inline README.md]: grammar

File: README.md [hunk 4]
```diff
@@ -1,55 +1,140 @@
...
+***Generative sampling***
+
+You can use the script: ```train_BUCTD_synthesis_noise.sh```.
+
+***Empirical sampling***
+
+You can match your own BU models by updating the scripts in [./data_preprocessing/](./data_preprocessing/). 
 
-We are grateful to the authors of [HRNet](https://github.com/HRNet/deep-high-resolution-net.pytorch) and [MIPNet](https://rawalkhirodkar.github.io/mipnet) as our code builds on their excellent work. 
+If you don't want to match your own BU models for training, we provide the training annotations. You can download the annotations [here](https://drive.google.com/drive/folders/17UnDWUtvcXrmuH90_wYrDUxYDOBkQ0C1?usp=share_link).
```
Comment (MMathisLab) [inline README.md:83]: BU needs defined

File: README.md [hunk 5]
```diff
@@ -1,55 +1,140 @@
...
 
+During inference, we use different BU/One-stage model's predictions (e.g. PETR, CID) as Conditions. The result files can be downloaded from the link above.
```
Comment (MMathisLab) [inline README.md]: why is One captialized?

File: README.md [hunk 6]
```diff
@@ -1,55 +1,140 @@
...
-We are grateful to the authors of [HRNet](https://github.com/HRNet/deep-high-resolution-net.pytorch) and [MIPNet](https://rawalkhirodkar.github.io/mipnet) as our code builds on their excellent work. 
+If you don't want to match your own BU models for training, we provide the training annotations. You can download the annotations [here](https://drive.google.com/drive/folders/17UnDWUtvcXrmuH90_wYrDUxYDOBkQ0C1?usp=share_link).
 
+During inference, we use different BU/One-stage model's predictions (e.g. PETR, CID) as Conditions. The result files can be downloaded from the link above. 
 
+</details>
+
+### Testing
+<details>
+  <summary>We also provide the best model per human dataset along with the testing scripts.</summary>
```
Comment (MMathisLab) [inline README.md:91]: define  "best" highest mAP on what benchmark?

File: README.md [hunk 7]
```diff
@@ -1,55 +1,140 @@
...
+
+### Code Acknowledgements
+
+We are grateful to the authors of [HRNet](https://github.com/HRNet/deep-high-resolution-net.pytorch), [MIPNet](https://rawalkhirodkar.github.io/mipnet), and [TransPose](https://github.com/yangsenius/TransPose) as our code builds on their excellent work. 
 
 ## Reference
 
 If you find this code or ideas presented in our work useful, please cite:
 
 [Rethinking pose estimation in crowds: overcoming the detection information-bottleneck and ambiguity](https://arxiv.org/abs/2306.07879) by Mu Zhou*, Lucas Stoffl*, Mackenzie W. Mathis and Alexander Mathis.
```
Comment (MMathisLab) [inline README.md]: Update citation to ICCV2023

**Review by MMathisLab — CHANGES_REQUESTED** [review]
- first pass on grammar

**Comment by LucZot** [comment]
> * first pass on grammar

Thanks for the comments, README is now updated

File: README.md [hunk 8]
```diff
@@ -1,55 +1,140 @@
...
-
-- Stand-alone code will be made available by ICCV (Oct 2023)!
+<img src="media/BUCTD_fig1.png" width="600">
 
 - This code will also be integrated in [DeepLabCut](https://github.com/DeepLabCut/DeepLabCut)!
 
-### Code Acknowledgements
+### Installation 
+
+We developed and tested our models under ```python=3.8.10, pytorch=1.8.0, cuda=11.1```. Other versions may work just as well.
```
Comment (MMathisLab) [inline README.md]: under --> with

Comment (MMathisLab) [reply README.md]: "Other versions may also be suitable"

File: README.md (hunk 3 above)
Comment (MMathisLab) [reply README.md]: @LucZot this isn't fixed "install other needed packages --> "Install additional dependencies"

File: README.md (hunk 4 above)
Comment (MMathisLab) [reply README.md:83]: Not fixed? @LucZot

**Review by MMathisLab — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (LucZot) [reply README.md:83]: It is now defined in the sentence before

**Comment by LucZot** [comment]
Addresses #1

**Review by MMathisLab — APPROVED** [review]

**Pull request merged**